tests/integration-service/namespace-backed-environments.go

```diff
@@ -34,224 +33,171 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(f.AsKubeAdmin.CommonController.HaveTestsSucceeded(snapshot)).To(BeTrue(), fmt.Sprintf("tests have not succeeded for snapshot %s/%s", snapshot.GetNamespace(), snapshot.GetName()))
 			})
 
-			It("should find the related Integration Test PipelineRun", func() {
-				timeout = time.Minute * 5
-				interval = time.Second * 2
-				Eventually(func() error {
-					testPipelinerun, err = f.AsKubeAdmin.IntegrationController.GetIntegrationPipelineRun(integrationTestScenario.Name, snapshot.Name, testNamespace)
-					if err != nil {
-						GinkgoWriter.Printf("failed to get Integration test PipelineRun for a snapshot '%s' in '%s' namespace: %+v\n", snapshot.Name, testNamespace, err)
-						return err
-					}
-					if !testPipelinerun.HasStarted() {
-						return fmt.Errorf("pipelinerun %s/%s hasn't started yet", testPipelinerun.GetNamespace(), testPipelinerun.GetName())
-					}
-					return nil
-				}, timeout, interval).Should(Succeed(), fmt.Sprintf("timed out when waiting for the Integration PipelineRun to start for the IntegrationTestScenario/Snapshot : %s/%s", integrationTestScenario.Name, snapshot.Name))
-				Expect(testPipelinerun.Labels["appstudio.openshift.io/snapshot"]).To(ContainSubstring(snapshot.Name))
-				Expect(testPipelinerun.Labels["test.appstudio.openshift.io/scenario"]).To(ContainSubstring(integrationTestScenario.Name))
-				Expect(testPipelinerun.Labels["appstudio.openshift.io/environment"]).To(ContainSubstring(ephemeralEnvironment.Name))
+			It("should lead to SnapshotEnvironmentBinding getting deleted", func() {
+				Eventually(func() bool {
+					snapshotEnvironmentBinding, err = f.AsKubeAdmin.CommonController.GetSnapshotEnvironmentBinding(applicationName, testNamespace, ephemeralEnvironment)
+					return snapshotEnvironmentBinding == nil
+				}, time.Minute*3, time.Second*2).Should(BeTrue(), fmt.Sprintf("timed out when waiting for SnapshotEnvironmentBinding to be deleted for application %s/%s", testNamespace, applicationName))
 			})
 
-			When("Integration Test PipelineRun is created", func() {
-				It("should eventually complete successfully", func() {
-					Expect(f.AsKubeAdmin.IntegrationController.WaitForIntegrationPipelineToBeFinished(integrationTestScenario, snapshot, testNamespace)).To(Succeed(), fmt.Sprintf("Error when waiting for a integration pipeline for snapshot %s/%s to finish", testNamespace, snapshot.GetName()))
-				})
+			It("should lead to ephemeral environment getting deleted", func() {
+				Eventually(func() bool {
+					ephemeralEnvironment, err = f.AsKubeAdmin.GitOpsController.GetEphemeralEnvironment(snapshot.Spec.Application, snapshot.Name, integrationTestScenario.Name, testNamespace)
+					return ephemeralEnvironment == nil
+				}, time.Minute*3, time.Second*1).Should(BeTrue(), fmt.Sprintf("timed out when waiting for the Ephemeral Environment %s to be deleted", ephemeralEnvironment.Name))
```

Comment:
@psturc I was thinking about logging the `err` variable within `fmt.Sprintf()` function so that it tells us what exactly went wrong. Currently, we only rely on this error string (`"timed out when waiting..."`) to understand what went wrong in case of failures. Wdyt?

If you agree, how can I log this `err` variable? Should I directly use it within `fmt.Sprintf()`? I doubt it might contain the value from previous `It()/Wehn()` blocks...

Reply:
wdyt about this approach?
```go
Eventually(func() error {
    ephemeralEnvironment, err = f.AsKubeAdmin.GitOpsController.GetEphemeralEnvironment(snapshot.Spec.Application, snapshot.Name, integrationTestScenario.Name, testNamespace)
    return err
}, time.Minute*3, time.Second*1).Should(Succeed(), fmt.Sprintf("timed out when waiting for the Ephemeral Environment %s to be deleted", ephemeralEnvironment.Name))
```

if `Eventually` fails on timeout, it logs the assertion details, including the whole error message, like this:
```
Expected "error message bla bla"
to be: nil
```

Reply:
Thank you, this approach looks great, I'll update the other `Eventually` block!

Also, in this specific `Eventually` block, rather than nil `err` we should be aiming to get a non-nil `err`, because here we're trying to assert the absence of ephemeral env.

However, a non-nil `err` would signify either an ephemeral env is deleted or there was an error during listing the environments ([from here](https://github.com/redhat-appstudio/e2e-tests/pull/763/files#diff-83ad1c2d880481b63484c7f59c8da9a6fb630c54db48ebfbfe1c407583889272R44-R54)), which could be misleading. Wdyt?

Reply:
> Also, in this specific Eventually block, rather than nil err we should be aiming to get a non-nil err, because here we're trying to assert the absence of ephemeral env.

oh, right, I missed that.

would this work then?
```go
Eventually(func() error {
    ephemeralEnvironment, err = f.AsKubeAdmin.GitOpsController.GetEphemeralEnvironment(snapshot.Spec.Application, snapshot.Name, integrationTestScenario.Name, testNamespace)
    return err
}, time.Minute*3, time.Second*1).ShouldNot(Succeed()), fmt.Sprintf("timed out when waiting for the Ephemeral Environment %s to be deleted", ephemeralEnvironment.Name))
Expect(err).To(ContainSubstring("no matching ephemeral environment found"))
```
ideally we should have the "no matching ephemeral environment found" error message stored in `pkg/contants`, so in case it changes or is removed, the compiler detects that.

Reply:
Thanks!
Yes, this works! Updated. Please check again.